### PR TITLE
Logger#close nilable error fix

### PR DESCRIPTION
**Pull request by Exilor**
And a spec that would have revealed the bug earlier.

### CODE REVIEW

**Comment by BlaXpirit** [comment]
Seems like the check fails because the automatic tool thinks the code should be formatted differently. Please run `crystal tool format` and amend the commit(s).

File: spec/std/logger_spec.cr [hunk 1]
```diff
@@ -86,4 +86,11 @@ describe "Logger" do
     logger.info { a = 1 }
     a.should eq(0)
   end
+
```
Comment (BlaXpirit) [inline spec/std/logger_spec.cr]: Pretty sure it's these two spaces alone on the line

**Comment by Exilor** [comment]
Good catch @BlaXpirit , I ran the tool on my computer and obviously I didn't *see* any changes so I thought it was ok :smile:

**Comment by BlaXpirit** [comment]
It would be best to not introduce a 3rd commit just with the fix. Maybe you meant to amend the commit but something went wrong.

`rebase -i HEAD~2` with changing the 2nd commit to `fixup` can help.

**Comment by Exilor** [comment]
Yeah, every time I type ```git [something]``` god kills a kitten. But it should be ok now.

File: spec/std/logger_spec.cr [hunk 2]
```diff
@@ -86,4 +86,11 @@ describe "Logger" do
...
+  it "closes" do
+    IO.pipe do |r, w|
+      Logger.new(w).close
+      w.closed?.should be_true
+    end
```
Comment (ysbaddaden) [inline spec/std/logger_spec.cr]: Maybe test logger too?

```crystal
IO.pipe do |_, io|
  logger = Logger.new(io)
  logger.close
  logger.closed?.should be_true
  io.closed?.should be_true
end
```

Comment (Exilor) [inline spec/std/logger_spec.cr]: But Logger itself doesn't have a ```closed?``` method.

**Comment by ysbaddaden** [comment]
Oops. My bad.

**Pull request merged**